migrations/000021_permissions_db.up.sql

```diff
@@ -0,0 +1,80 @@
+BEGIN;
+    CREATE SCHEMA IF NOT EXISTS permissions;
+    SET search_path = permissions, public, pg_catalog;
+
+    --
+    -- Table for storing the types of resources to which permissions may be applied.
+    --
+    CREATE TABLE IF NOT EXISTS resource_types (
+        id uuid NOT NULL DEFAULT uuid_generate_v1(),
+        name varchar(64) NOT NULL,
+        description text NOT NULL,
+        PRIMARY KEY (id)
+    );
+
+    --
+    -- Uniqueness constraint on the resource type name.
+    --
+    CREATE UNIQUE INDEX IF NOT EXISTS resource_types_name_unique
+        ON resource_types (lower(trim(regexp_replace(name, '\s+', ' ', 'g'))));
```

Comment:
This has to be done as a unique index because of the functions (or at least, I couldn't figure out a way to do it with an inline UNIQUE)